sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -83,9 +86,12 @@ joint_matrix_apply(Group sg, joint_matrix<Group, T, Use, M, N, Layout> &jm,
 #if defined(__SYCL_DEVICE_ONLY__)
 #if defined(__NVPTX__)
   std::ignore = sg;
-  for (int i = 0; i < jm.cuda_impl.wi_marray.size(); i++) {
-    lambda(jm.cuda_impl.wi_marray[i]);
+  for (int i = 0; i < jm.matrix_impl.wi_marray.size(); i++) {
```

Comment:
Similarly here, if you match the marray names in cuda/hip impls, this can be simplified.

Reply:
that should be done for `copy` as well.